### ISSUE TITLE
[Trying to add a subscription] This package or the Stripe doc is not correct.

### ISSUE THREAD

**Issue by samvoults** [stripe/stripe-react-native#1523]
I'm trying to add Stripe and allow users to first give their payment info on the Payment Element sheet, then click on the subscribe button to start their subscription (with a free trial).

I am following a tutorial from the Stripe docs and at this step (https://stripe.com/docs/payments/accept-a-payment-deferred?platform=react-native&type=payment#initialize-paymentsheet), they provide a code snippet that calls initPaymentSheet(...) without a client secret.

```
const { error } = await initPaymentSheet({
  merchantDisplayName: "Example, Inc.",
  intentConfiguration: {
    mode: {
      amount: 6099,
      currencyCode: 'USD',
    },
    confirmHandler: handleConfirmation
  }
});
```

And when I use the their code, I got the error: `You must provide either paymentIntentClientSecret or setupIntentClientSecret.`

The tutorial from the Stripe doc doesn't mention to add a secret client at this step but mention it only after so I don't undestand why I got this error.

Is something missing in the Stripe docs? Is it this package? Or should I never touch a keyboard again?

**Comment by seanzhang-stripe** [comment]
Hi @samvoults This flow requires a more recent version of `stripe-react-native` SDK. Can you update to the latest version and try again?

**Comment by samvoults** [comment]
Thanks for you answer. I did it and it worked until the next step.

Do you know who wrote that tutorial (stripe.com/docs/payments/accept-a-payment-deferred?platform=react-native&type=payment)?

I don't understand why there are 2 price amounts ("6099" client side then "1099" server side).

Also the tutorial mentions to pass the paymentMethod.id to the server but then in the code snippet to create the payment intent on the server, there is no use of paymentMethod.id...

**Comment by seanzhang-stripe** [comment]
Thanks for the feedback, I'd ask the docs team to fix it. Let me know if you have any other questions.

**Comment by seanzhang-stripe** [comment]
Hi @samvoults I saw an inconsistency where you are using [PaymentMode](https://stripe.dev/stripe-react-native/api-reference/modules/PaymentSheet.html#PaymentMode) at the client side, but your backend creates a SetupIntent. You should always use `SetupMode` with SetupIntent, or `PaymentMode` with PaymentIntent.

There's a dedicated integration [guide](https://stripe.com/docs/billing/subscriptions/build-subscriptions?ui=elements) for Subscription integration that I strongly recommend you to go through. Although it's not tailored for react-native, the basic idea here is to use a `PaymentMode` PaymentSheet to pay the first invoice generated by the subscription. 

Feel free to let me know if you have any other questions.

**Comment by samvoults** [comment]
Hello @seanzhang-stripe, I have followed the Subscription integration guide.

The problem I have is when the user presses on the "Subscribe" button, the subscription and the free trial are directly started even though the user hasn't entered any card info.

Here is my simple code client side :

```
// 1. First the user presses on the "Subscribe" button.
const onSubscribeButtonPress = async () => {
  await initializePaymentSheet();
  const { error } = await presentPaymentSheet();
  console.log("error", error);
};

// 2. Then the PaymentSheet is initialized.
const initializePaymentSheet = async () => {
  const response = await fetch(`${API_URL}/create-subscription`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const { clientSecret } = await response.json();
  const { error } = await initPaymentSheet({
    setupIntentClientSecret: clientSecret,
    returnURL: "partyfinder://stripe-redirect",
    merchantDisplayName: "Example, Inc.",
  });
  console.log("error", error);
};
```

Server side:
```
// 3. The subscription is created.
app.post("/test", async (req, res) => {
  const subscription = await stripe.subscriptions.create({
    customer: "cus_OcolLDEHx71Deg",
    items: [
      {
        price: "price_1NnKx1DS664Ek1kLvTiUghvb",
      },
    ],
    trial_period_days: 1,
    payment_behavior: "default_incomplete",
    payment_settings: { save_default_payment_method: "on_subscription" },
    expand: ["latest_invoice.payment_intent"],
  });

  const setupIntent = await stripe.setupIntents.retrieve(
    subscription.pending_setup_intent
  );

  res.send({
    clientSecret: setupIntent.client_secret,
  });
});
```

The scenario I'm trying to implement is: The user presses on the "Subscribe" button, he writes his card info and then, when he presses on the "Pay" button, the subscription with the free trial start.

Thank you for your time.

**Comment by YoucefBen47** [comment]
@samvoults Hi, have you fixed the issue where the free trial is initially active just after the payment sheet is open with no card details entered ?

**Comment by YoucefBen47** [comment]
@seanzhang-stripe Hi,
I have also posted an issue about the issue where a subscription that has a free trial is set to active after the payment sheet is opened without even filling out the card details. Every time a user opens the payment sheet it creates an active subscription that has no card attached to it yet.

**Comment by YoucefBen47** [comment]
@seanzhang-stripe The issue is some users keep opening and closing the payment sheet which causes the creation of multiple free trials with the same user.

**Comment by YoucefBen47** [comment]
@samvoults  Thank you and I hope it'll work out for you.

**Comment by seanzhang-stripe** [comment]
Hi @YoucefBen47 @samvoults In this case you should defer the subscription creation.
1.  Create a SetupIntent,  and use PaymentSheet to collect a payment method from the customer
2.  If there's no error from [presentPaymentSheet](https://stripe.dev/stripe-react-native/api-reference/index.html#presentPaymentSheet), your react-native app should call [retrieveSetupIntent](https://stripe.dev/stripe-react-native/api-reference/index.html#retrieveSetupIntent) and get its `payment_method`
3.  Pass the `payment_method` to your backend and use it as the `default_payment_method` when creating a subscription.

**Comment by YoucefBen47** [comment]
@seanzhang-stripe thank you for replying but the issue is not with creating the payment intent. The issue is that each time the user opens the payment sheet there's an active subscription created with no card details. We have instances where 6 subscriptions are created and it's really messing up our metrics. Is there a way to prevent a free trial from being active unless the user fills the card details and clicks on the Setup button ?

**Comment by seanzhang-stripe** [comment]
@YoucefBen47 I don't think presenting a PaymentSheet would have a side effect of creating a subscription. Can you share with me the relevant code,  as well as the ID of the subscription so that I can take a look at?

**Comment by YoucefBen47** [comment]
@seanzhang-stripe I even tested the repo from the official video tutorials. Please test here this is Charlie's repo.
https://github.com/charliecruzan-stripe/react-native-video-series
He has a setup intent example for a trial subscription. You will notice that once you click to initialise the payment sheet a subscription would have already been created and set to Paid without even entering the card details or clicking on the Setup button on the payment sheet. 
If a user keeps opening and closing the payment sheet it will keep creating more subscriptions that have no cards attached to it.
In the attached image the text says the subscription was marked as paid because the amount due was 0.00$ however it is set to paid and active trial without a card.
<img width="1399" alt="Screenshot 2023-10-11 at 6 21 20 AM" src="https://github.com/stripe/stripe-react-native/assets/70854871/40d77e01-b7c9-4d04-95ab-19b4e6248241">

**Comment by YoucefBen47** [comment]
@seanzhang-stripe here are two screenshots to highlight Charlie's code. Once you land on the set up intent example page the `fetchPaymentSheetParams` is fired and it creates an active paid free trial without even clicking on the `Subscribe to monthly delivery button`.
<img width="1440" alt="Screenshot 2023-10-11 at 6 31 53 AM" src="https://github.com/stripe/stripe-react-native/assets/70854871/a92f9938-7b53-468e-b00f-e60a29956bb5">
<img width="1440" alt="Screenshot 2023-10-11 at 6 32 19 AM" src="https://github.com/stripe/stripe-react-native/assets/70854871/7e0d694a-b739-4b47-a889-c4984dfd7f48">

**Comment by YoucefBen47** [comment]
@samvoults @seanzhang-stripe So what we were doing was initialising the subscription first then updating it after the user submits the card details. Now I use a setup intent first to collect the card details and then create a trial after the user clicks to setup an intent. The subscription will use the default payment method that was collected from the setup intent. I will test to make sure the payment is taken from the card when the trial is over.

**Comment by woerchri** [comment]
@seanzhang-stripe: i also have the same problem regarding the intentConfiguration. It also throws an error saying i need a payment or setup intent. I am on version 0.33.0.. Is there any recommendation to get it to work or am i using the wrong version?

**Comment by seanzhang-stripe** [comment]
@YoucefBen47 Sounds like a plan! Let me know if you have any follow-up questions! 

@samvoults I believe your initial question is already answered. I'll close this thread for now, feel free to re-open it or create a new ticket if you need more help!
